actor-share-conversion/src/gf2_128/sender.rs

```diff
@@ -0,0 +1,229 @@
+use super::{A2MMessage, M2AMessage, SendTapeMessage};
+use mpc_aio::protocol::ot::{OTFactoryError, ObliviousSend};
+use mpc_core::{ot::config::OTSenderConfig, Block};
+use share_conversion_aio::{
+    gf2_128::{
+        recorder::{Recorder, Tape, Void},
+        Gf2ConversionMessage, SendTape, Sender as IOSender,
+    },
+    AdditiveToMultiplicative, MultiplicativeToAdditive, ShareConversionError,
+};
+use share_conversion_core::gf2_128::Gf2_128ShareConvert;
+use utils_aio::{adaptive_barrier::AdaptiveBarrier, factory::AsyncFactory, mux::MuxChannelControl};
+use xtra::prelude::*;
+
+enum State<
+    T: AsyncFactory<OT>,
+    OT: ObliviousSend<[Block; 2]>,
+    U: Gf2_128ShareConvert,
+    V: MuxChannelControl<Gf2ConversionMessage>,
+    W: Recorder<U>,
+> {
+    Initialized {
+        id: String,
+        barrier: Option<AdaptiveBarrier>,
+        muxer: V,
+        sender_factory: T,
+    },
+    Setup(IOSender<T, OT, U, W>),
+    Complete,
+}
+
+#[derive(xtra::Actor)]
+pub struct Sender<T, OT, U, V, W = Void>
+where
+    T: AsyncFactory<OT>,
+    OT: ObliviousSend<[Block; 2]>,
+    U: Gf2_128ShareConvert,
+    V: MuxChannelControl<Gf2ConversionMessage>,
+    W: Recorder<U>,
+{
+    state: State<T, OT, U, V, W>,
+}
+
+impl<T, OT, U, V, W> Sender<T, OT, U, V, W>
+where
+    T: AsyncFactory<OT>,
+    OT: ObliviousSend<[Block; 2]>,
+    U: Gf2_128ShareConvert,
+    V: MuxChannelControl<Gf2ConversionMessage>,
+    W: Recorder<U>,
+{
+    pub fn new(id: String, barrier: Option<AdaptiveBarrier>, muxer: V, sender_factory: T) -> Self {
+        Self {
+            state: State::Initialized {
+                id,
+                barrier,
+                muxer,
+                sender_factory,
+            },
+        }
+    }
+}
+
+impl<T, OT, U, V, W> Sender<T, OT, U, V, W>
+where
+    T: AsyncFactory<OT, Config = OTSenderConfig, Error = OTFactoryError> + Send,
+    OT: ObliviousSend<[Block; 2]> + Send,
+    U: Gf2_128ShareConvert + Send,
+    V: MuxChannelControl<Gf2ConversionMessage>,
+    W: Recorder<U>,
+{
+    pub async fn setup(&mut self) -> Result<(), ShareConversionError> {
+        // We need to own the state, so we use this only as a temporary modification
+        let state = std::mem::replace(&mut self.state, State::Complete);
```

Comment:
shouldn't this be explicitely `State::Error` instead of`State::Complete`? Otherwise this causes the code reviewer to pause and wonder if there may be security implications.

Reply:
yes. ::Temporary is fine